components/server/src/workspace/workspace-starter.ts

```diff
@@ -950,38 +949,15 @@ export class WorkspaceStarter {
         }
     }
 
-    private async tryEnableConnectionLimiting(
-        featureFlags: NamedWorkspaceFeatureFlag[],
-        user: User,
-        billingTier: BillingTier,
-    ) {
-        const wsConnectionLimitingEnabled = await getExperimentsClientForBackend().getValueAsync(
-            "workspace_connection_limiting",
-            false,
-            {
-                user,
-                billingTier,
-            },
-        );
-
-        if (wsConnectionLimitingEnabled) {
-            const shouldLimitNetworkConnections = await this.entitlementService.limitNetworkConnections(
-                user,
-                new Date(),
-            );
-            if (shouldLimitNetworkConnections) {
-                featureFlags.push("workspace_connection_limiting");
-            }
+    private async tryEnableConnectionLimiting(featureFlags: NamedWorkspaceFeatureFlag[], user: User) {
```

Comment:
[nit] this function (and the other one too) performs side-effects on the feature-flags object supplied. Would you be able to adjust it that instead it returns whether it should be enabled, and the value of the flag and the flag is appended as part of the parent call-flow, rather than through a side-effect? It makes for much more readable (and expectable) code.

Alternatively, return a new featureflags object and re-assign such that it at least returns something